federation/pkg/federation-controller/service/servicecontroller.go

```diff
@@ -728,8 +728,12 @@ func (s *ServiceController) lockedUpdateDNSRecords(service *cachedService, clust
 	for key := range s.clusterCache.clientMap {
 		for _, clusterName := range clusterNames {
 			if key == clusterName {
-				s.ensureDnsRecords(clusterName, service)
-				ensuredCount += 1
+				err := s.ensureDnsRecords(clusterName, service)
+				if err != nil {
+					glog.Errorf("Failed to ensure DNS records for service %v in cluster %s due to %v", service, clusterName, err)
+				} else {
+					ensuredCount += 1
+				}
 			}
 		}
 	}
```

Comment:
The error 2 lines below is not correct now. We need to keep track of number of missing cluster clients and number of failed ensureDNsRecords and return an error accordingly.